lib/devicetrust/enroll/os_darwin.go

```diff
@@ -0,0 +1,21 @@
+// Copyright 2022 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package enroll
+
+import devicepb "github.com/gravitational/teleport/api/gen/proto/go/teleport/devicetrust/v1"
+
+func getDeviceOSType() devicepb.OSType {
```

Comment:
Isn't this just a very roundabout way to get GOOS? Why not a GOOS to `devicepb.OSType` function instead?

Reply:
Build tags are logic-free, but I suppose a GOOS solution is easier to read. Changed.